test/group_pool_wf_test.go

```diff
@@ -0,0 +1,328 @@
+// Group tests
+// ------------
+//    1. Get group info
+//    2. validate group merge
+//    3. Do group merge
+//    4. modify default_iscsi_target_scope
+// Pool tests
+// -----------
+//   1. Get pool details - get pools name
+//   2. Merge two pools
+//   3. Edit Pool - remove array from pool
+//   4. Create a new pool with available / above removed pool
+//   5. Change pool name
+//   6. Create volume in new pool
+//   7. Delete pool when volumes are present
+//       Should fail, validate error msg
+//   8. Delete volume from new pool
+//   9. Delete pool
+//       Should be deleted
+
+package test
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/client/v1/nimbleos"
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/param"
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/service"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type GroupPoolWorkflowSuite struct {
+	suite.Suite
+	groupService            *service.NsGroupService
+	arraygroupService       *service.GroupService
+	volumeService           *service.VolumeService
+	poolService             *service.PoolService
+	arrayService            *service.ArrayService
+	sourcearrayGroupService *service.GroupService
+}
+
+func (suite *GroupPoolWorkflowSuite) SetupSuite() {
+	groupService, err := config()
+	assert.Nilf(suite.T(), err, "Could not connect to array")
+	suite.groupService = groupService
+	suite.arraygroupService = groupService.GetGroupService()
+	suite.poolService = groupService.GetPoolService()
+	suite.arrayService = groupService.GetArrayService()
+	suite.volumeService = groupService.GetVolumeService()
+}
+
+func (suite *GroupPoolWorkflowSuite) connectSourceArray() *service.NsGroupService {
+	srcgroupService, err := service.NewNsGroupService(sourceArrayIP, sourceArrayusername, sourceArraypassword, "v1", true)
+	assert.Nilf(suite.T(), err, "Could not connect to array")
+	srcgroupService.SetDebug()
+	return srcgroupService
+}
+
+// Get Leader array pool details
+func (suite *GroupPoolWorkflowSuite) TestGetGroupDetailsLeaderArray() {
+	filter := &param.GetParams{}
+	groupResp, err := suite.arraygroupService.GetGroups(filter)
+	groupName := *groupResp[0].Name
+	fmt.Printf("source array group name %+v\n", groupName)
+	assert.Nilf(suite.T(), err, "Get Group failed")
+	_, err = suite.arraygroupService.GetGroupByName(groupName)
+	assert.Nilf(suite.T(), err, "Get Group by name failed")
+}
+
+// Validate & Performe Group Merge
```

Comment:
Makes sense. Later we would be adding tests for replication as-well, which will require multi arrays. We'll come-up with some instructions &  an option which would skip running such tests. For now we can keep the name as 'group_pool_wf_xtest.go', so it will skip this test file.